packages/plugin-analytics/src/utils/PlausibleAnalytics.ts

```diff
@@ -0,0 +1,100 @@
+import { logger } from 'robo.js'
+import { BaseEngine, EventOptions, ViewOptions } from './analytics'
+
+export class PlausibleAnalytics extends BaseEngine {
+	private _PLAUSIBLE_DOMAIN = process.env.PLAUSIBLE_DOMAIN
+
+	public async view(page: string, options: ViewOptions): Promise<void> {
+		const temp = {
+			name: 'pageview',
+			url: options?.url ?? `https://${this._PLAUSIBLE_DOMAIN}${page}`,
+			domain: options?.domain ?? this._PLAUSIBLE_DOMAIN
+		}
+
+		if (typeof options.data === 'object') {
+			if (options.data !== null) {
+				if (Object.entries(options.data).length > 30) {
+					throw new Error('[Plausible] Cannot send an object with more than 30 fields.')
+				} else {
+					Object.assign(temp, { props: { ...options.data } })
+				}
+			}
+		}
+
+		const res = await fetch('https://plausible.io/api/event', {
```

Comment:
Please include the Plausible API domain as a global. Not important, but it may help~

```ts
// Global scope
const PlausibleDomain = 'https://plausible.io'
```

Then:
```ts
fetch(PlausibleDomain + '/api/event', {
```